### PR TITLE
Duality of cells

### PR DESCRIPTION
**Pull request by thibautbenjamin**
Compute the opposite of a cell in a weak omega-catt (cf duality paper with Ioannis)

This branch is based on top of #46, and should be rebased there

### CODE REVIEW

**Comment by thibautbenjamin** [comment]
@regular-citizen ready to review after #46. definition and testing for opposites. Straightforwardly following the article

**Comment by regular-citizen** [comment]
Having trouble reviewing this because the diff is based on master somehow, so edits are there twice with the refactoring branch

File: lib/opposite.ml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+let rec ty typ op_data =
+  let d = Unchecked.dim_ty typ in
+  match typ with
+  | Obj -> Obj
+  | Arr(a,t,u) ->
+    let a = ty a op_data in
+    let t = tm t op_data in
+    let u = tm u op_data in
+    if List.mem d op_data then Arr(a,u,t) else Arr(a,t,u)
+  | Meta_ty m -> Meta_ty m
```
Comment (regular-citizen) [inline lib/opposite.ml:47]: Why is the meta_type unchanged by taking the opposite?

Comment (regular-citizen) [reply lib/opposite.ml:47]: As discussed, not needed because constraints are not yet applied at this stage

**Pull request merged**